### PR TITLE
Closes #102: ADLB vignette

### LINKED ISSUE #102: Feature Request:  ADLB vignette
### Feature Idea

LB vignette to accompany LB dataset

### Relevant Input

**General algorithms / composite score** 
* FSI (LB: GGT + Triglycerides; VS: BMI + Waist circumference)
* HSI (LB: ALT + AST; ADMH/ADSL: Diabetes status; ADSL: Sex)

For these, we would need to have code that 1) merges a dataset/datasets, 2) calculates score, and 3) drop dataset in the vignette.

Alternative could be function, but that could be a lot of work.

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by starosto**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral<ext> (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by starosto** [comment]
Hi @pharmaverse/admiralmetabolic,

I have loaded the ADLB vignette code and a draft function for consideration if we want to proceed with a function-based approach or not.

We now create 4 parameters: HSI, FLI, NAFLD, and HOMA-IR. This can be discussed if we need all of them, as the logic is similar across these scores.

1) Issues with `pharmaversesdtm::lb_metabolic`:
    - ALT (for HSI, NAFLD) and Platelets (for NAFLD) are missing. For the time being, I've assigned these randomly.
    - Unit conversion is needed for:
        - Triglycerides: mmol/L to mg/dL (for FLI)
        - Albumin: g/L to g/dL (for NAFLD)
        - Insulin: pmol/L to mIU/L (for HOMA-IR)
        
        Do we want to handle these unit conversions or update the units at the SDTM level?
    - `lb_metabolic` is a grouped dataframe, I had to ungroup manually.

2) The `data` folder does not contain `admiralmetabolic_advs` and `admiralmetabolic_adsl`.

3) The `admiral_adsl` dataset does not contain T2DM variable. I've assigned this randomly for now.

4) Currently, there are two ways for deriving metabolic parameters: a manual approach and using functions. We need to decide whether to create functions, and if so, what the best solution would be:
    - one function vs four separate functions (one for each parameter)?
    - data merging step (ADVS, ADSL) occur within the function(s) or outside?
    - assignment of PRAMN/PARAMCD/PARAM occur within the function(s) or outside?

    Please see the sections `## Derive Metabolic Parameters` vs `## Derive Metabolic Parameters using Functions` in the ADLB vignette code. I have no experience in writing functions in R, so I appreciate your support 👍

**Review by AndersAskeland — COMMENTED** [review]
Amazing start @starosto . 

Overall I think the vignette is looking good. I do however think we need an overall discussion  on how we want to proceed with the vignette. 

Referring back to the #102 about how we should handle data coming from different domains. I was originally a proponent of having functionalities that can be used independent of execution order of ADaM programs, however, have lately been back-paddling a bit, now leaning more towards defining an execution order, and keeping the traceability of variables.

I am also leaning towards not having functions, but instead showing the calculations in the vignette.

Let's discuss it at our next meeting.

**Comment by AndersAskeland** [comment]
@pharmaverse/admiralmetabolic Any additional inputs here? I think most of the next meeting will be used discussing how we proceed with the vignette, so would be great if people had a look at it 👍

File: vignettes/adlb.Rmd [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+-   [Example Script](#example)
+
+## Read in Data {#readdata}
+
+To start, all data frames needed for the creation of the ADaM dataset should be loaded into the global environment. Reading data will usually be a company specific process, however, for the purpose of this vignette, we will use example data from `{pharmaversesdtm}` and `{admiral}`. We will utilize `DM`, `LB` and `ADSL`.
+
+```{r, message=FALSE, warning=FALSE}
+dm_metabolic <- pharmaversesdtm::dm_metabolic
+lb_metabolic <- pharmaversesdtm::lb_metabolic # Use lb_metabolic
+admiral_adsl <- admiral::admiral_adsl
```
[the remark this answers is not in the document]
Comment (starosto) [reply vignettes/adlb.Rmd:65]: will update once #73 is closed

File: vignettes/adlb.Rmd [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+  mutate(TRT01P = ARM, TRT01A = ACTARM) %>%
+  derive_vars_merged(
+    dataset_add = admiral_adsl,
+    by_vars = get_admiral_option("subject_keys"),
+    new_vars = exprs(TRTSDT, TRTEDT)
+  )
+
+# Keep only required ADSL variables for derivations
+adsl_vars <- exprs(TRTSDT, TRTEDT, TRT01A, TRT01P)
+```
```
[the remark this answers is not in the document]
Comment (starosto) [reply vignettes/adlb.Rmd:87]: will update once #73 is closed

File: NEWS.md [hunk 3]
```diff
@@ -39,3 +39,4 @@
 
 - Vignette "Creating a Metabolic ADVS ADaM". (#28) 
 - Vignette "Creating a Control of Eating Questionnaire ADaM". (#53)
+- Vignette "Creating a Metabolic ADLB ADaM". (#102)
```
[the remark this answers is not in the document]
Comment (starosto) [reply NEWS.md:42]: is any action required on my part?

File: vignettes/adlb.Rmd [hunk 4]
```diff
@@ -0,0 +1,495 @@
...
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT))
+
+adlb <- adlb %>%
+  ## Add PARAMCD PARAM and PARAMN - from LOOK-UP table ----
+  derive_vars_merged_lookup(
+    dataset_add = param_lookup,
+    new_vars = exprs(PARAMCD, PARAM, PARAMN),
+    by_vars = exprs(LBTESTCD),
+    check_type = "none", # Set to "warning" or "error" if you want checks
+    print_not_mapped = FALSE # Set to TRUE to see unmapped LBTESTCDs
```
[the remark this answers is not in the document]
Comment (starosto) [reply vignettes/adlb.Rmd:136]: I think previously I added missing ALT and Platelets to the param_lookup and I had a warning, I will use defaults

File: vignettes/adlb.Rmd [hunk 5]
```diff
@@ -0,0 +1,495 @@
...
+      PARAMCD == "INSULMIU" ~ 12
+    )
+  )
+
+# Combine original ADLB with converted records
+adlb <- bind_rows(adlb, adlb_converted) %>%
+  arrange(STUDYID, USUBJID, ADT, PARAMN) # Arrange for consistency
+```
+
+## Merge ADVS and ADSL variables {#merge}
```
[the remark this answers is not in the document]
Comment (starosto) [reply vignettes/adlb.Rmd:253]: I see in [Derive Additional Covariates from VS and LB](https://pharmaverse.github.io/admiral/articles/pk_adnca.html#addcovar) they merge SDTM data (and use [compute_egfr](https://pharmaverse.github.io/admiral/reference/compute_egfr.html) function). I was trying to follow this logic but using ADVS and ADSL. Now I understand suggestion from @yurovska [issue #102](https://github.com/pharmaverse/admiralmetabolic/issues/102#issuecomment-2789259398) to maybe consider additional dataset for this purpose. I do not know what the best solution for the package would be....

File: vignettes/adlb.Rmd [hunk 6]
```diff
@@ -0,0 +1,495 @@
...
+adlb_random_lab <- adlb %>%
+  select(STUDYID, USUBJID, AVISIT, AVISITN, ADT, ADY) %>%
+  distinct() %>%
+  mutate(
+    AVAL = case_when(
+      TRUE ~ round(runif(n(), min = 10, max = 50), 1) # Random ALT values
+    ),
+    PARAMCD = "ALT",
+    PARAM = "Alanine Aminotransferase (U/L)",
+    PARAMN = 13
```
[the remark this answers is not in the document]
Comment (starosto) [reply vignettes/adlb.Rmd:268]: If we want to derive HSI we need to add ALT.
If we want to derive NAFLD we need to add ALT and Platelets.

File: vignettes/adlb.Rmd [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+      TRUE ~ LBSTRESN
+    ),
+    # Only populate AVALC if character value is non-redundant with AVAL, except for TRIG and INSULIN
+    AVALC = case_when(
+      LBTESTCD %in% c("TRIG", "INSULIN") ~ NA_character_,
+      is.na(AVAL) | as.character(AVAL) != LBSTRESC ~ LBSTRESC, # Compare AVAL with LBSTRESC
+      TRUE ~ NA_character_
+    ),
+    ANRLO = LBSTNRLO,
+    ANRHI = LBSTNRHI
```
Comment (yurovska) [inline vignettes/adlb.Rmd:152]: @starosto Ref Ranges are now incorrect for `TRIG` and `INSULIN` that got CV units after your last update but LLN/ULN remain in SI.

File: vignettes/adlb.Rmd [hunk 8]
```diff
@@ -0,0 +1,324 @@
...
+  "ALB", "ALB", "Albumin (g/L)", 1,
+  "ALP", "ALKPH", "Alkaline Phosphatase (U/L)", 2,
+  "AST", "AST", "Aspartate Aminotransferase (U/L)", 3,
+  "CHOL", "CHOLES", "Cholesterol (mmol/L)", 4,
+  "GGT", "GGT", "Gamma Glutamyl Transferase (U/L)", 5,
+  "GLUC", "GLUC", "Glucose (mmol/L)", 6,
+  "HBA1CHGB", "HBA1CHGB", "Hemoglobin A1C/Hemoglobin (%)", 7,
+  "INSULIN", "INSULIN", "Insulin (mIU/L)", 8,
+  "TRIG", "TRIG", "Triglycerides (mg/dL)", 9
+)
```
Comment (yurovska) [inline vignettes/adlb.Rmd:104]:
```suggestion
param_lookup <- tibble::tribble(
  ~LBTESTCD, ~PARAMCD, ~PARAM, ~PARAMN, ~PARCAT2,
  "ALB", "ALB", "Albumin (g/L)", 1, "SI",
  "ALP", "ALKPH", "Alkaline Phosphatase (U/L)", 2, "SI",
  "AST", "AST", "Aspartate Aminotransferase (U/L)", 3, "SI",
  "CHOL", "CHOLES", "Cholesterol (mmol/L)", 4, "SI",
  "GGT", "GGT", "Gamma Glutamyl Transferase (U/L)", 5, "SI",
  "GLUC", "GLUC", "Glucose (mmol/L)", 6, "SI",
  "HBA1CHGB", "HBA1CHGB", "Hemoglobin A1C/Hemoglobin (%)", 7, "CV",
  "INSULIN", "INSULIN", "Insulin (mIU/L)", 8, "CV",
  "TRIG", "TRIG", "Triglycerides (mg/dL)", 9, "CV"
)
```

It would be better to define `PARCAT2` in the lookup table instead of deriving it conditionally.

File: vignettes/adlb.Rmd [hunk 9]
```diff
@@ -0,0 +1,324 @@
...
+    new_vars_prefix = "A",
+    dtc = LBDTC
+  ) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT))
+  
+ adlb <- adlb %>%
+  ## Add PARAMCD PARAM and PARAMN - from LOOK-UP table
+  derive_vars_merged_lookup(
+    dataset_add = param_lookup,
+    new_vars = exprs(PARAMCD, PARAM, PARAMN),
```
Comment (yurovska) [inline vignettes/adlb.Rmd:130]:
```suggestion
    new_vars = exprs(PARAMCD, PARAM, PARAMN, PARCAT2),
```

File: vignettes/adlb.Rmd [hunk 10]
```diff
@@ -0,0 +1,324 @@
...
+    by_vars = exprs(LBTESTCD)
+  ) %>%
+  ## Calculate PARCAT1 AVAL AVALC ANRLO ANRHI
+  mutate(
+    PARCAT1 = LBCAT,
+    PARCAT2 = case_when(
+      LBTESTCD %in% c("TRIG", "INSULIN") ~ "CV",
+      TRUE ~ "SI"
+    ),
```
Comment (yurovska) [inline vignettes/adlb.Rmd:139]:
```suggestion
```

**Review by yurovska — CHANGES_REQUESTED** [review]

File: vignettes/adlb.Rmd (hunk 8 above)
Comment (yurovska) [reply vignettes/adlb.Rmd:104]: By the way, I've noticed that HbA1c (%) is in CV units too. The standard (SI) units are actually `mmol/mol`. I think this is a mistake in `pharmaversesdtm::lb_metabolic`.

File: vignettes/adlb.Rmd (hunk 7 above)
Comment (yurovska) [reply vignettes/adlb.Rmd:152]: See https://github.com/pharmaverse/admiralmetabolic/pull/110#discussion_r2106798928

File: vignettes/adlb.Rmd (hunk 8 above)
Comment (starosto) [reply vignettes/adlb.Rmd:104]: I decided to derive PARCAT2 conditionally in slice_derivation(), please let me know if something would be against it? 
As for HbA1c units, I think we **can/should(?)** have data in both units:
https://loinc.org/59261-8/
_-All new instruments sold after January 1st, 2011 will report (as a result of an Hgb A1c test) **both SI** (mmol/mol - no decimals) **and NGSP** derived units (percentage - one decimal), in agreement with the Consensus Statement._

But then I am not sure if PARCAT2="SI" is good for HbA1c [%], @pharmaverse/admiralmetabolic let us know what your opinion is here

File: vignettes/adlb.Rmd [hunk 11]
```diff
@@ -0,0 +1,324 @@
...
+### Homeostasis Model Assessment – Insulin Resistance (HOMA-IR)
+
+This part describes how to calculate the HOMA-IR index from fasting plasma glucose and fasting plasma insulin. This derivation does not require merging data from other datasets.
+
+```{r}
+# Derive HOMA-IR using derive_param_computed
+adlb <- adlb %>%
+  derive_param_computed(
+    by_vars = exprs(!!!get_admiral_option("subject_keys"), AVISIT, AVISITN, ADT, ADY, !!!adsl_vars),
+    parameters = c("INSULIN", "GLUC"),
```
[the remark this answers is not in the document]
Comment (starosto) [reply vignettes/adlb.Rmd:268]: this is my assumption; maybe actually having LBFAST is the standard for fasting tests? @pharmaverse/admiralmetabolic please comment on this

File: vignettes/adlb.Rmd (hunk 8 above)
Comment (yurovska) [reply vignettes/adlb.Rmd:104]: "SI" is not good for HbA1c in %, it must be "CV". Moreover, `LBSTRESU = "%"` is not good in `pharmaversesdtm::lb_metabolic` as it's not the standard units for it. Not sure whether we want to fix it in `pharmaversesdtm`, so at least here let's assign a proper categorization.

File: vignettes/adlb.Rmd (hunk 11 above)
[the remark this answers is not in the document]
Comment (yurovska) [reply vignettes/adlb.Rmd:268]: It would be nice to have `LBFAST` in `pharmaversesdtm::lb_metabolic` for consistency.

File: vignettes/adlb.Rmd (hunk 8 above)
Comment (manciniedoardo) [reply vignettes/adlb.Rmd:104]: Update from today's call: agreement to change SI unit to mmol/mol in LB in pharmaversesdtm, then CV unit should be re-derived in ADLB as % if required. Issue to be made in pharmaversesdtm in order to update this. Then PARCAT2 should be assigned to SI. 

https://github.com/pharmaverse/pharmaversesdtm/issues/176

Comment (manciniedoardo) [reply vignettes/adlb.Rmd:104]: Can resolve this conversation once the PR in pharmaversesdtm is merged and change is done in ADLB vignette.

Comment (yurovska) [reply vignettes/adlb.Rmd:104]: The PR in `pharmaversesdtm` is already there waiting for review/approval. I've kept HbA1c in `%` (LBORRES) and converted to `mmol/mol` (LBSTRESN), so both options are available.

@starosto It's up to you to decide which one to pick up. I would personally go ahead with `mmol/mol` to be in line with Glucose in `mmol/L`. I'm just used to see them in pair, so either both Glucose and HbA1c in SI units or both in CV.

**Comment by AndersAskeland** [comment]
So, I ended up skipping doing reviews, but both `ADSL` and and `admiralmetabolic_vs` is now merged. Is we needing anything else to complete this @starosto ?

**Comment by starosto** [comment]
Hi @AndersAskeland, I am currently using the following assignments in my code:
```R
admiralmetabolic_adsl <- admiralmetabolic::admiralmetabolic_adsl
advs <- admiralmetabolic::admiralmetabolic_advs
```
Could you please confirm that `adsl_metabolic` should be used instead of `admiralmetabolic_adsl`?
In the data folder, I see `admiralmetabolic_advs` but no `adsl_metabolic`. Is this intentional?

**Comment by yurovska** [comment]
@AndersAskeland ADSL dataset has not even been committed, do it's not there under https://github.com/pharmaverse/admiralmetabolic/tree/main/data

**Comment by AndersAskeland** [comment]
Sorry, forgot to add the file to generate the adsl dataset. It's in #106

**Comment by AndersAskeland** [comment]
@starosto There is now data 👍

**Comment by starosto** [comment]
hi @AndersAskeland and @yurovska and @pharmaverse/admiralmetabolic members, last chance to change something, please review and comment. If all good, I think you can merge @AndersAskeland

**Comment by AndersAskeland** [comment]
@starosto Amazing! I will look at it tomorrow 👍

**Review by yurovska — APPROVED** [review]
LGTM! Thanks @starosto !

File: vignettes/adlb.Rmd [hunk 12]
```diff
@@ -0,0 +1,324 @@
...
+        # Only populate AVALC if character value is non-redundant with AVAL
+        AVALC = if_else(
+          is.na(AVAL) | as.character(AVAL) != LBSTRESC,
+          LBSTRESC,
+          NA_character_
+        ),
+        ANRLO = LBSTNRLO,
+        ANRHI = LBSTNRHI
+      )
+    )
```
Comment (AndersAskeland) [inline vignettes/adlb.Rmd:152]: Are there anything keeping us from having everything as SI units for AVAL, and then use LBORRES directly for the calculations of HOMA-IR and FLI?

File: vignettes/adlb.Rmd [hunk 13]
```diff
@@ -0,0 +1,324 @@
...
+
+```{r}
+# Derive HOMA-IR using derive_param_computed
+adlb <- adlb %>%
+  derive_param_computed(
+    by_vars = exprs(!!!get_admiral_option("subject_keys"), AVISIT, AVISITN, ADT, ADY, !!!adsl_vars),
+    parameters = c("INSULIN", "GLUC"),
+    set_values_to = exprs(
+      AVAL = if_else(
+        complete.cases(AVAL.INSULIN, AVAL.GLUC),
```
Comment (AndersAskeland) [inline vignettes/adlb.Rmd:258]: Will not `derive_vars_computed` automatically handle this? 

I've been doing some testing, and I think we can keep it as:

```r
 AVAL = AVAL.INSULIN * AVAL.GLUC / 22.5,
 
```

File: vignettes/adlb.Rmd [hunk 14]
```diff
@@ -0,0 +1,324 @@
...
+
+```{r}
+# Derive FLI using derive_param_computed
+adlb <- adlb %>%
+  derive_param_computed(
+    by_vars = exprs(!!!get_admiral_option("subject_keys"), AVISIT, AVISITN, ADT, ADY, BMI, WSTCIR, !!!adsl_vars),
+    parameters = c("TRIG", "GGT"),
+    set_values_to = exprs(
+      AVAL = if_else(
+        complete.cases(AVAL.TRIG, BMI, AVAL.GGT, WSTCIR),
```
Comment (AndersAskeland) [inline vignettes/adlb.Rmd:290]: Same as above, I don't think we need the `if_else(complete.cases...`

File: vignettes/adlb.Rmd [hunk 15]
```diff
@@ -0,0 +1,324 @@
...
+# Example display of derived data
+dataset_vignette(
+  arrange(adlb, USUBJID, PARCAT1, ADY, PARAMN),
+  display_vars = exprs(!!!get_admiral_option("subject_keys"), PARAMCD, PARAM, PARCAT1, AVAL, ADY, AVISIT, AVISITN)
+)
+```
+
+For deriving visits based on time-windows, see `{admiral}` [Visit and Period Variables](https://pharmaverse.github.io/admiral/articles/visits_periods.html).
+
+## Metabolic Parameters - Description {#metabolic_params_desc}
```
Comment (AndersAskeland) [inline vignettes/adlb.Rmd:195]:
```suggestion
## Derive Metabolic Parameters {#metabolic_params_desc}
```

File: vignettes/adlb.Rmd [hunk 16]
```diff
@@ -0,0 +1,324 @@
...
+*   **Homeostasis Model Assessment – Insulin Resistance (HOMA-IR)**: Calculated from fasting plasma glucose (mmol/L) and fasting plasma insulin (mIU/L).
+ $$ \text{HOMA-IR} = \frac{FPI \times FPG}{22.5}. $$
+An alternative formula is used when fasting plasma glucose is assessed in mg/dL: $$ \text{HOMA-IR} = \frac{FPI \times FPG}{405}. $$
+
+To derive these parameters and scores, the following variables will be needed:
+
+*   From ADLB: TRIG (Triglycerides), GGT, GLUC (fasted), INSULIN (fasted).
+*   From ADVS: BMI, WSTCIR (Waist Circumference).
+
+## Merge ADVS variables {#merge}
```
Comment (AndersAskeland) [inline vignettes/adlb.Rmd:213]:
```suggestion
### Add variables required for derivation {#merge}
```

File: vignettes/adlb.Rmd [hunk 17]
```diff
@@ -0,0 +1,324 @@
...
+In this section, we will merge relevant variables from the ADVS (Analysis Dataset for Vital Signs) dataset into the ADLB dataset. This is necessary to include variables such as BMI and Waist Circumference, which are required for the derivation of FLI score parameter in subsequent section.
```
Comment (AndersAskeland) [inline vignettes/adlb.Rmd:214]:
```suggestion
In this section, we will merge relevant variables needed to derive metabolic parameter from the ADVS (Analysis Dataset for Vital Signs) dataset into the ADLB dataset. This is necessary to include variables such as BMI and Waist Circumference, which are required for the derivation of FLI score parameter in subsequent section.
```
```suggestion
In this section, we will merge relevant variables from the ADVS (Analysis Dataset for Vital Signs) dataset into the ADLB dataset. This is necessary to include variables such as BMI and Waist Circumference, which are required for the derivation of FLI score parameter in subsequent section.
```

File: vignettes/adlb.Rmd [hunk 18]
```diff
@@ -0,0 +1,324 @@
...
+dataset_vignette(
+  arrange(adlb, USUBJID, ADY, PARAMN),
+  display_vars = exprs(!!!get_admiral_option("subject_keys"), PARAMCD, PARAM, AVAL, ADT, ADY, AVISIT, AVISITN, BMI, WSTCIR)
+)
+```
+
+## Derive Metabolic Parameters {#derive_metabolic}
+
+In this section, we will derive the metabolic parameters and scores described previously by directly implementing the calculation formulas.
+
```
Comment (AndersAskeland) [inline vignettes/adlb.Rmd:245]:
```suggestion
```

File: NEWS.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply NEWS.md:48]: We still need to move this. If you want I can do it right before merging?

File: vignettes/adlb.Rmd [hunk 19]
```diff
@@ -0,0 +1,324 @@
...
+```{r echo=FALSE}
+# Example display of derived data
+dataset_vignette(
+  filter(adlb, PARAMCD == "HOMAIR") %>%
+    arrange(USUBJID, ADY, PARAMN),
+  display_vars = exprs(!!!get_admiral_option("subject_keys"), PARAMCD, PARAM, AVAL, ADY, AVISIT, AVISITN)
+)
+```
+
+### Fatty Liver Index (FLI)
```
Comment (AndersAskeland) [inline vignettes/adlb.Rmd:278]:
```suggestion
### Fatty Liver Index (FLI) {#derive_fli}
```

File: vignettes/adlb.Rmd [hunk 20]
```diff
@@ -0,0 +1,324 @@
...
+  arrange(adlb, USUBJID, ADY, PARAMN),
+  display_vars = exprs(!!!get_admiral_option("subject_keys"), PARAMCD, PARAM, AVAL, ADT, ADY, AVISIT, AVISITN, BMI, WSTCIR)
+)
+```
+
+## Derive Metabolic Parameters {#derive_metabolic}
+
+In this section, we will derive the metabolic parameters and scores described previously by directly implementing the calculation formulas.
+
+### Homeostasis Model Assessment – Insulin Resistance (HOMA-IR)
```
Comment (AndersAskeland) [inline vignettes/adlb.Rmd:246]:
```suggestion
### Homeostasis Model Assessment – Insulin Resistance (HOMA-IR) {#derive_homa}
```

File: vignettes/adlb.Rmd [hunk 21]
```diff
@@ -0,0 +1,324 @@
...
+
+# Programming Workflow
+
+-   [Read in Data](#readdata)
+-   [Define Lookup Tables](#lookups)
+-   [Derive Core ADLB Variables](#derive_core)
+-   [Derive Visit Information](#derive_visits)
+-   [Metabolic Parameters - Description](#metabolic_params_desc)
+-   [Merge ADVS variables](#merge)
+-   [Derive Metabolic Parameters](#derive_metabolic)
```
Comment (AndersAskeland) [inline vignettes/adlb.Rmd:52]:
```suggestion
-   [Derive Metabolic Parameters](#metabolic_params_desc)
    -   [Merge ADVS variables](#merge)
    -   [Derive HOMA index](#derive_homa)
    -   [Derive FLI score](#derive_FLI)
```

File: vignettes/adlb.Rmd (hunk 12 above)
Comment (starosto) [reply vignettes/adlb.Rmd:152]: @AndersAskeland I think it is also ok to have CV units, especially if they are used in further derivations
